File: week_8_and_9/3_Ruby/fizz_buzz/my_solution.rb
# U3.W8-9: 


# I worked on this challenge [by myself, with: ].

# 2. Pseudocode

# define method super_fizzbuzz that takes in an array as a parameter
# 

# 3. Initial Solution

def super_fizzbuzz(array)
	new_array = []
	array.each do |num| 
		case num
		when num % 15 == 0 then new_array << "FizzBuzz"
		when num % 3 == 0 then new_array << "Fizz"
		when num % 5 == 0 then new_array << "Buzz"
		else new_array << num
		end
	end
	puts new_array
end



# 4. Refactored Solution






# 1. DRIVER TESTS/ASSERT STATEMENTS GO BELOW THIS LINE

puts "returns 'Fizz' for multiples of 3"
puts super_fizzbuzz([3,6,9]) == ["Fizz", "Fizz", "Fizz"]

puts "returns 'Buzz' for multiples of 5"
puts super_fizzbuzz([5,10,15]) == ["Buzz", "Buzz", "Buzz"]

puts "returns 'FizzBuzz' for multiples of 15"
puts super_fizzbuzz([15,30,45]) == ["FizzBuzz", "FizzBuzz", "FizzBuzz"]

puts "works on [1,2,3]"
puts super_fizzbuzz([1,2,3]) == [1,2,'Fizz']

puts "works on [15, 5, 3, 1]"
puts super_fizzbuzz([15,5,3,1]) == ['FizzBuzz', 'Buzz', 'Fizz', 1]



# 5. Reflection 

File: week_8_and_9/2_JavaScript/grocery_list/my_solution.js
// U3.W8-9: 


// I worked on this challenge [by myself, with: ].


// User Stories (As many as you want. Delete the statements you don't need)
// As a user, I want to be able to add an item 
// As a user, I want to be able to puts all items on list
// As a user, I want to be able to physically remove an item
// As a user, I want to be able to swipe it all clean 
// As a user, I want to be able to change amount of items
 
// 2. Pseudocode



// 3. Initial Solution

var groceryList = {};

groceryList.list = {};

groceryList.addItem = function (item, num) {
	groceryList.list[item] = num;
};

groceryList.removeItem = function (item) {
	delete groceryList.list[item];
};

groceryList.clearList = function () {
	groceryList.list = {};
};

groceryList.changeAmount = function (item, num) {
	groceryList.list[item] = num;
};

groceryList.printList = function () {
	console.log(groceryList.list);
};

// 4. Refactored Solution



// 1. DRIVER TESTS/ASSERT STATEMENTS GO BELOW THIS LINE

groceryList.addItem("banana", 5);
groceryList.addItem("apple", 6);
groceryList.addItem("eggs", 1);
console.log(groceryList.list);

groceryList.removeItem("banana");
console.log(groceryList.list);

groceryList.changeAmount("apples", 2);
groceryList.printList();
groceryList.clearList();
groceryList.printList();
// groceryList.clearList();


// 5. Reflection 